gapis/shadertools/shadertools.go

```diff
@@ -189,6 +189,35 @@ func descriptorBindingLess(a DescriptorBinding, b DescriptorBinding) bool {
 	return a.SpirvId < b.SpirvId
 }
 
+func ParseSourceCode(shader []uint32) (string, error) {
+	spvReflectErr := func(res C.SpvReflectResult) error {
+		if res == C.SPV_REFLECT_RESULT_SUCCESS {
+			return nil
+		}
+		return fmt.Errorf("SPIRV-Reflect failed with error code %v\n", res)
+	}
+
+	module := C.SpvReflectShaderModule{}
+
+	shaderPtr := unsafe.Pointer(nil)
+	if len(shader) > 0 {
+		shaderPtr = unsafe.Pointer(&shader[0])
+	}
+	if err := spvReflectErr(C.spvReflectCreateShaderModule(
+		C.size_t(len(shader)*4),
+		shaderPtr,
+		&module)); err != nil {
+		return "", err
+	}
+	defer C.spvReflectDestroyShaderModule(&module)
+
+	if module.source_source == nil {
+		return "", fmt.Errorf("No Source File Detected\n")
```

Comment:
remove the `\n`. Also use `errors.New("...")`, since you aren't actually formatting anything.